pinecone/models.go

```diff
@@ -32,31 +32,31 @@ const (
 )
 
 type IndexStatus struct {
-	Ready bool
-	State IndexStatusState
+	Ready bool             `json:"ready"`
+	State IndexStatusState `json:"state"`
 }
 
 type IndexSpec struct {
-	Pod        *PodSpec
-	Serverless *ServerlessSpec
+	Pod        *PodSpec        `json:"pod,omitempty"`
+	Serverless *ServerlessSpec `json:"serverless,omitempty"`
 }
 
 type Index struct {
-	Name      string
-	Dimension int32
-	Host      string
-	Metric    IndexMetric
-	Spec      *IndexSpec
-	Status    *IndexStatus
+	Name      string       `json:"name"`
+	Dimension int32        `json:"dimension"`
+	Host      string       `json:"host"`
+	Metric    IndexMetric  `json:"metric"`
+	Spec      *IndexSpec   `json:"spec,omitempty"`
+	Status    *IndexStatus `json:"status,omitempty"`
 }
 
 type Collection struct {
-	Name        string
-	Size        *int64
-	Status      CollectionStatus
-	Dimension   *int32
-	VectorCount *int32
-	Environment string
+	Name        string           `json:"name"`
+	Size        *int64           `json:"size,omitempty"`
+	Status      CollectionStatus `json:"status"`
+	Dimension   *int32           `json:"dimension,omitempty"`
+	VectorCount *int32           `json:"vector_count,omitempty"`
```

Comment:
This might be an exception to the `omitempty` thing I was saying above -- if `vector_count` is empty, that actually is meaningful, right? Like, that means there are no vectors in the collection, right? If so, I think we should actually _not_ have `omitempty` there, since the value of `0` indicates the actual state of things (vs an ambiguous state)

Reply:
The main reason for using `omitempty` here is `VectorCount` is a pointer, and I've tried to use `omitempty` with all our pointers. The reason is otherwise an empty pointer is marshaled to `null` in the JSON results. 

For collections specifically, this is mainly to cover a specific issue with `VectorCount` and other fields possibly being empty when creating a collection from an empty index.

Reply:
I reworked the `Collection` struct and the `toCollection` function to map the pointers into zero values for the ints to make things easier to work with for consumers.